### ISSUE TITLE
Microsoft.Web/sites/extensions VS2015 Errors

### ISSUE THREAD

**Issue by toddbrownmcis** [Azure/azure-resource-manager-schemas#120]
Should the **connectionString** property support an empty string?
Should the **setParameters** property support expression values?

```
"name": "MSDeploy",
"type": "extensions",
"location": "[resourceGroup().location]",
"apiVersion": "2015-08-01",
"properties": {
  "connectionString": "",
  "setParameters": {
    "IIS Web Application Name": "[variables('appServiceName')]"
  }
```

Both of these are reported as schema errors in VS2015

```
String must be at least 1 characters long and at most 2147483647
Value must be one of the following types: string

```

**Comment by yangl900** [comment]
Hi @davidebbo could you help have someone take a look?

**Comment by davidebbo** [comment]
@shobak101 is the expert.

But note that the VS schema is likely outdated, so just because it gives warnings does not imply that the template will fail to deploy.

**Comment by shobak101** [comment]
for the first point, my question is why set an empty connectionstring? if no DB is involved you can add a property "dbType": "None" and that is it. 
for the second point: yes expression variables are supported in setParameters, but in this template appServiceName is a parameter right? why try using a variable for it?

**Comment by toddbrownmcis** [comment]
Removing the empty connectionString works fine with `"dbType": "None"`.

schemas/**2014-06-01**/Microsoft.Web.json supports setParameters expressions but schemas/**2015-08-01**/Microsoft.Web.json does not.

**Comment by bmoore-msft** [comment]
@shobak101 - I think setParamters needs to be an array right?  Isn't that what's allowed there?  I can set any arbitrary parameters needed by my msdeploy package. Right now it's defined as a "string" which is clearly not correct give what actually works during deployment

**Comment by ifeoluwaokunoren** [comment]
setParameters is fixed in later api-version:
https://github.com/Azure/azure-resource-manager-schemas/blob/ff320d13e002510dc247f13440a3e70fa70ad479/schemas/2018-02-01/Microsoft.Web.json#L4980-L4994

There is also no validation on connectionString now, so empty string should be fine:
https://github.com/Azure/azure-resource-manager-schemas/blob/ff320d13e002510dc247f13440a3e70fa70ad479/schemas/2018-02-01/Microsoft.Web.json#L4968-L4971
